<launch>
	<!-- middle top and bottom -->
	<!-- <node pkg="tf" type="static_transform_publisher" name="dep_cam_top_link_to_base_link" 
		args="0.2 0 0.355 0 -0.3007058 0 0.953717 base_link dep_cam_top_link 100" />
	<node pkg="tf" type="static_transform_publisher" name="dep_cam_bottom_link_to_base_link" 
		args="0.2 0 0.315 0 0.3007058 0 0.953717 base_link dep_cam_bottom_link 100" /> -->

	<!-- left(top) and right(bottom) -->
	<node pkg="tf" type="static_transform_publisher" name="dep_cam_top_link_to_base_link" 
		args="0.17 0.12 0.355 0.1270838 -0.4030582 0.272532 0.8643611 base_link dep_cam_top_link 100" />
	<node pkg="tf" type="static_transform_publisher" name="dep_cam_bottom_link_to_base_link" 
		args="0.17 -0.12 0.355 -0.1270838 -0.4030582 -0.272532 0.8643611 base_link dep_cam_bottom_link 100" />

	<node pkg="tf" type="static_transform_publisher" name="dep_cam_laser_link_to_base_link" 
		args="0 0 0.355 0 0 0 1 base_link dep_cam_laser_link 100" />
	
	<group ns="dep_cam_top">
		<node pkg="msa010_ros" type="msa010_ros_driver.py" name="msa010_ros_driver">
			<param name="device" value="/dev/depth_camera_top" />
			<param name="frame_id" value="dep_cam_top_link" />
			<param name="rot_img" value="-1" />
		</node>
		<node pkg="msa010_ros" type="depth_image_to_point_cloud_node" name="depth_image_to_point_cloud_node" output="screen" />
		<node pkg="msa010_ros" type="point_cloud_processing_node" name="point_cloud_processing_node">
			<param name="sub_topic" value="depth/points" />
			<param name="LeafSize" value="0.05" />
			<param name="MeanK" value="20" />
			<param name="StddevMulThresh" value="1.0" />
			<param name="RadiusSearch" value="0.1" />
			<param name="MinNeighborsInRadius" value="3" />
		</node>
		<node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan">
	        <remap from="cloud_in" to="processed_point_cloud" />
	        <remap from="scan" to="depth/scan" />
	        <param name="target_frame" value="dep_cam_laser_link" />
	        <param name="transform_tolerance" value="0.01" />
	        <param name="min_height" value="0.4" />
	        <param name="max_height" value="1.2" />
	        <param name="angle_min" value="0.0872665" />
	        <param name="angle_max" value="1.13446401" />
	        <param name="angle_increment" value="0.0122173048" />
	        <param name="scan_time" value="0.1" />
	        <param name="range_min" value="0.2" />
	        <param name="range_max" value="2.5" />
	        <param name="use_inf" value="true" />
	        <param name="concurrency_level" value="1" />
		</node>
	</group>

	<group ns="dep_cam_bottom">
		<node pkg="msa010_ros" type="msa010_ros_driver.py" name="msa010_ros_driver">
			<param name="device" value="/dev/depth_camera_bottom" />
			<param name="frame_id" value="dep_cam_bottom_link" />
			<param name="rot_img" value="-1" />
		</node>
		<node pkg="msa010_ros" type="depth_image_to_point_cloud_node" name="depth_image_to_point_cloud_node" output="screen" />
		<node pkg="msa010_ros" type="point_cloud_processing_node" name="point_cloud_processing_node">
			<param name="sub_topic" value="depth/points" />
			<param name="LeafSize" value="0.05" />
			<param name="MeanK" value="20" />
			<param name="StddevMulThresh" value="1.0" />
			<param name="RadiusSearch" value="0.1" />
			<param name="MinNeighborsInRadius" value="3" />
		</node>
		<node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan">
	        <remap from="cloud_in" to="processed_point_cloud" />
	        <remap from="scan" to="depth/scan" />
	        <param name="target_frame" value="dep_cam_laser_link" />
	        <param name="transform_tolerance" value="0.01" />
	        <param name="min_height" value="0.4" />
	        <param name="max_height" value="1.2" />
	        <param name="angle_min" value="-1.13446401" />
	        <param name="angle_max" value="-0.0872665" />
	        <param name="angle_increment" value="0.0122173048" />
	        <param name="scan_time" value="0.1" />
	        <param name="range_min" value="0.2" />
	        <param name="range_max" value="2.5" />
	        <param name="use_inf" value="true" />
	        <param name="concurrency_level" value="1" />
		</node>
	</group>

	<!-- <group ns="dep_cam_bottom">
		<node pkg="msa010_ros" type="msa010_ros_driver.py" name="msa010_ros_driver">
			<param name="device" value="/dev/depth_camera_bottom" />
			<param name="frame_id" value="dep_cam_bottom_link" />
			<param name="rot_img" value="-1" />
		</node>
		<node pkg="msa010_ros" type="cliff_detector.py" name="cliff_detector" output="screen">
			<param name="img_topic" value="depth/image_raw" />
			<param name="info_topic" value="depth/camera_info" />
			<param name="frame_id" value="dep_cam_bottom_link" />
			<param name="cam_height" value="0.315" />
			<param name="cam_angle" value="35" />
			<param name="cliff_threshold" value="0.1" />
			<param name="img_freq" value="10" />
			<param name="range_min" value="0.2" />
			<param name="range_max" value="2.5" />
			<param name="cam_x" value="0.2" />
			<param name="cam_y" value="0.0" />
			<param name="cam_z" value="0.315" />
			<param name="row_upper" value="50" />
			<param name="col_left" value="10" />
			<param name="col_right" value="90" />
			<param name="skip_row_upper" value="1" />
			<param name="skip_row_bottom" value="1" />
			<param name="tilt_compensation" value="3" />
		</node>
	</group> -->

<!-- 	<node pkg="tf" type="static_transform_publisher" name="dep_cam_top_link_to_base_link" 
		args="0.21 0 0.32 0 0.1736482 0 0.9848078 base_link dep_cam_top_link 100" />

	<group ns="dep_cam_top">
		<node pkg="msa010_ros" type="msa010_ros_driver.py" name="msa010_ros_driver">
			<param name="device" value="/dev/depth_camera_top" />
			<param name="frame_id" value="dep_cam_top_link" />
			<param name="rot_img" value="-1" />
		</node>
		<node pkg="msa010_ros" type="depth_image_to_point_cloud_node" name="depth_image_to_point_cloud_node" output="screen" />
		<node pkg="msa010_ros" type="point_cloud_processing_node" name="point_cloud_processing_node">
			<param name="sub_topic" value="depth/points" />
			<param name="LeafSize" value="0.05" />
			<param name="MeanK" value="20" />
			<param name="StddevMulThresh" value="1.0" />
			<param name="RadiusSearch" value="0.1" />
			<param name="MinNeighborsInRadius" value="3" />
		</node>
		<node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan">
	        <remap from="cloud_in" to="processed_point_cloud" />
	        <remap from="scan" to="depth/scan" />
	        <param name="target_frame" value="dep_cam_laser_link" />
	        <param name="transform_tolerance" value="0.01" />
	        <param name="min_height" value="0.0" />
	        <param name="max_height" value="0.2" />
	        <param name="angle_min" value="-0.610865238" />
	        <param name="angle_max" value="0.610865238" />
	        <param name="angle_increment" value="0.0122173048" />
	        <param name="scan_time" value="0.1" />
	        <param name="range_min" value="0.2" />
	        <param name="range_max" value="2.5" />
	        <param name="use_inf" value="true" />
	        <param name="concurrency_level" value="1" />
		</node>
	</group> -->

</launch>


